metrics/thresholds.go

```diff
@@ -188,6 +188,11 @@ func (ts *Thresholds) Run(sink Sink, duration time.Duration) (bool, error) {
 		ts.sinked["min"] = sinkImpl.Min
 		ts.sinked["max"] = sinkImpl.Max
 		ts.sinked["avg"] = sinkImpl.Avg
+
+		// Compute the median value of the trend sink
+		// We do this just in time, as it's needed, because it
+		// requires sorting the values, which computanionally expensive.
+		sinkImpl.Calc()
```

Comment:
I'm missing here a bit of the context, but for me, it sounds like without the `sinkImpl.Calc()` it's not safe to use the `sinkImpl.Med` which means that we can have a small getter in the `sinkImpl` which combines two operations and access to the `Med` should be done only by this getter. That way any other place where we potentially can read the value we don't forget to apply the calculations.

Reply:
I'm fine with the suggestions that @mstoykov did and also fine if the proper refactoring will be done in a separated PR